### PR TITLE
Enable exception tests on vert.x

### PR DESCRIPTION
**Pull request by laurit**

### CODE REVIEW

File: instrumentation/vertx-reactive-3.5/javaagent/src/test/groovy/server/VertxRxHttpServerTest.groovy [hunk 1]
```diff
@@ -53,25 +54,20 @@ class VertxRxHttpServerTest extends HttpServerTest<Vertx> implements AgentTestTr
...
   @Override
   String expectedServerSpanName(ServerEndpoint endpoint) {
-    return endpoint == PATH_PARAM ? "/path/:id/param" : endpoint.getPath()
+    switch (endpoint) {
+      case PATH_PARAM:
+        return "/path/:id/param"
+      case NOT_FOUND:
+        return "HTTP GET"
+    }
+    return endpoint.getPath()
```
Comment (iNikem) [inline instrumentation/vertx-reactive-3.5/javaagent/src/test/groovy/server/VertxRxHttpServerTest.groovy:70]: move this to `default:`?

**Review by iNikem — APPROVED** [review]

File: instrumentation/vertx-web-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/RoutingContextHandlerWrapper.java [hunk 2]
```diff
@@ -36,6 +40,24 @@ public void handle(RoutingContext context) {
     } catch (Exception ex) {
       log.error("Failed to update server span name with vert.x route", ex);
     }
-    handler.handle(context);
+    try {
+      handler.handle(context);
+    } catch (Throwable throwable) {
+      if (serverSpan != null && serverSpan.getSpanContext().isValid()) {
```
Comment (trask) [inline instrumentation/vertx-web-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/RoutingContextHandlerWrapper.java:46]: if it's not valid, I think span will be no-op anyways
```suggestion
      if (serverSpan != null) {
```

**Review by trask — APPROVED** [review]
👍

File: instrumentation/vertx-web-3.0/javaagent/src/test/groovy/server/VertxHttpServerTest.groovy [hunk 3]
```diff
@@ -48,30 +49,25 @@ class VertxHttpServerTest extends HttpServerTest<Vertx> implements AgentTestTrai
...
   @Override
   String expectedServerSpanName(ServerEndpoint endpoint) {
-    return endpoint == PATH_PARAM ? "/path/:id/param" : endpoint.getPath()
+    switch (endpoint) {
+      case PATH_PARAM:
+        return "/path/:id/param"
+      case NOT_FOUND:
+        return "HTTP GET"
+    }
+    return endpoint.getPath()
```
Comment (iNikem) [inline instrumentation/vertx-web-3.0/javaagent/src/test/groovy/server/VertxHttpServerTest.groovy:70]: Here too :)

**Review by anuraaga — APPROVED** [review]

**Pull request merged**